### PR TITLE
feat: set/upsert/remove returns WriteResult

### PR DESCRIPTION
**Pull request by Hieuzest**

### CODE REVIEW

File: packages/memory/src/index.ts [hunk 1]
```diff
@@ -150,18 +154,22 @@ export class MemoryDriver extends Driver {
 
   async upsert(sel: Selection.Mutable, data: any, keys: string[]) {
     const { table, model, ref } = sel
+    const result: Driver.WriteResult = { inserted: 0, modified: 0 }
```
Comment (MaikoTan) [inline packages/memory/src/index.ts:157]:
```suggestion
    const result = { inserted: 0, modified: 0 }
```

Then you could omit the `!` in the following code.

File: packages/sqlite/src/index.ts [hunk 2]
```diff
@@ -342,8 +342,9 @@ export class SQLiteDriver extends Driver {
   async remove(sel: Selection.Mutable) {
     const { query, table } = sel
     const filter = this.sql.parseQuery(query)
-    if (filter === '0') return
-    this.#run(`DELETE FROM ${escapeId(table)} WHERE ${filter}`)
+    if (filter === '0') return {}
+    const result = this.#run(`DELETE FROM ${escapeId(table)} WHERE ${filter}`, [], () => this.#get(`SELECT changes() AS id`))
+    return { removed: result.id }
```
Comment (MaikoTan) [inline packages/sqlite/src/index.ts:347]: Consider change the `id` to `count` or else more clear?

File: packages/sqlite/src/index.ts [hunk 3]
```diff
@@ -404,8 +406,9 @@ export class SQLiteDriver extends Driver {
   }
 
   async upsert(sel: Selection.Mutable, data: any[], keys: string[]) {
-    if (!data.length) return
+    if (!data.length) return {}
     const { model, table, ref } = sel
+    const result: Driver.WriteResult = { inserted: 0, modified: 0 }
```
Comment (MaikoTan) [inline packages/sqlite/src/index.ts:411]:
```suggestion
    const result = { inserted: 0, modified: 0 }
```

Similar to the one above.

File: packages/tests/src/update.ts [hunk 4]
```diff
@@ -133,12 +133,12 @@ namespace OrmOperations {
...
         return data.id
       })
-      await database.set('temp2', {
+      await expect(database.set('temp2', {
         $or: [
           { id: magicIds },
           { timestamp: magicBorn },
         ],
-      }, { list: ['2', '3', '3'] })
+      }, { list: ['2', '3', '3'] })).to.eventually.deep.equal({ modified: 3 })
```
Comment (MaikoTan) [inline packages/tests/src/update.ts:141]: `eventually` should be omitted here and everywhere?

**Review by MaikoTan — COMMENTED** [review]
Minor comments, also fine to ignore me.

File: packages/tests/src/update.ts (hunk 4 above)
Comment (Hieuzest) [reply packages/tests/src/update.ts:141]: can't omit

Comment (Hieuzest) [reply packages/tests/src/update.ts:141]: they are all promises

Comment (MaikoTan) [reply packages/tests/src/update.ts:141]: Ah...

**Review by shigma — APPROVED** [review]

**Pull request merged**